flux_led/__main__.py

```diff
@@ -810,6 +969,7 @@ def _read_msg(self, expected):
                 pass
             finally:
                 self._socket.setblocking(1)
+        print('recieving ', rx.hex())
```

Comment:
here is the code 

`    def _determine_query_len(self):

        # determine the type of protocol based of first 2 bytes.
        self._send_msg(bytearray([0x81, 0x8a, 0x8b]))
        rx = self._read_msg(2)
        # if any response is recieved, use the default protocol
        if len(rx) == 2:
            self._query_len = 14
            return
        # if no response from default recieved, next try the original protocol
        self._send_msg(bytearray([0xef, 0x01, 0x77]))
        rx = self._read_msg(2)
        if rx[1] == 0x01:
            self.protocol = 'LEDENET_ORIGINAL'
            self._use_csum = False
            self._query_len = 11
 
    def query_state(self, retry=2, led_type = None):

        if self._query_len == 0:
            self._determine_query_len()
            
        # default value
        msg = bytearray([0x81, 0x8a, 0x8b])
        # alternative for original protocol
        if self.protocol == 'LEDENET_ORIGINAL' or led_type == 'LEDENET_ORIGINAL':
            msg =  bytearray([0xef, 0x01, 0x77])
            led_type = 'LEDENET_ORIGINAL'

        try:
            self._send_msg(msg)
            rx = self._read_msg(self._query_len)
        except socket.error:
            if retry < 1:
                self._is_on = False
                return
            self.connect()
            return self.query_state(max(retry-1, 0), led_type)
        if rx is None or len(rx) < self._query_len:
            if retry < 1:
                self._is_on = False
                return rx
            return self.query_state(max(retry-1, 0), led_type)
        return rx
`